processor/memorylimiterprocessor/factory.go

```diff
@@ -48,57 +57,76 @@ func createDefaultConfig() config.Processor {
 	}
 }
 
-func createTracesProcessor(
+func (f *factory) createTracesProcessor(
 	_ context.Context,
 	set component.ProcessorCreateSettings,
 	cfg config.Processor,
 	nextConsumer consumer.Traces,
 ) (component.TracesProcessor, error) {
-	ml, err := newMemoryLimiter(set, cfg.(*Config))
+	memLimiter, err := f.getMemoryLimiter(set, cfg)
 	if err != nil {
 		return nil, err
 	}
 	return processorhelper.NewTracesProcessor(
 		cfg,
 		nextConsumer,
-		ml.processTraces,
+		memLimiter.processTraces,
 		processorhelper.WithCapabilities(processorCapabilities),
-		processorhelper.WithStart(ml.start),
-		processorhelper.WithShutdown(ml.shutdown))
+		processorhelper.WithStart(memLimiter.start),
+		processorhelper.WithShutdown(memLimiter.shutdown))
 }
 
-func createMetricsProcessor(
+func (f *factory) createMetricsProcessor(
 	_ context.Context,
 	set component.ProcessorCreateSettings,
 	cfg config.Processor,
 	nextConsumer consumer.Metrics,
 ) (component.MetricsProcessor, error) {
-	ml, err := newMemoryLimiter(set, cfg.(*Config))
+	memLimiter, err := f.getMemoryLimiter(set, cfg)
 	if err != nil {
 		return nil, err
 	}
 	return processorhelper.NewMetricsProcessor(
 		cfg,
 		nextConsumer,
-		ml.processMetrics,
+		memLimiter.processMetrics,
 		processorhelper.WithCapabilities(processorCapabilities),
-		processorhelper.WithShutdown(ml.shutdown))
+		processorhelper.WithStart(memLimiter.start),
+		processorhelper.WithShutdown(memLimiter.shutdown))
 }
 
-func createLogsProcessor(
+func (f *factory) createLogsProcessor(
 	_ context.Context,
 	set component.ProcessorCreateSettings,
 	cfg config.Processor,
 	nextConsumer consumer.Logs,
 ) (component.LogsProcessor, error) {
-	ml, err := newMemoryLimiter(set, cfg.(*Config))
+	memLimiter, err := f.getMemoryLimiter(set, cfg)
 	if err != nil {
 		return nil, err
 	}
 	return processorhelper.NewLogsProcessor(
 		cfg,
 		nextConsumer,
-		ml.processLogs,
+		memLimiter.processLogs,
 		processorhelper.WithCapabilities(processorCapabilities),
-		processorhelper.WithShutdown(ml.shutdown))
+		processorhelper.WithStart(memLimiter.start),
+		processorhelper.WithShutdown(memLimiter.shutdown))
+}
+
+func (f *factory) getMemoryLimiter(set component.ProcessorCreateSettings, cfg config.Processor) (*memoryLimiter, error) {
```

Comment:
Add some comments to explain the purpose of this function.

Reply:
👍